egs/ami/s5c/run.sh

```diff
@@ -0,0 +1,165 @@
+# 2. Training an overlap detector (using annotations) and corresponding
+# inference on full recordings.
+
+# We do not provide training script for an x-vector extractor. You
+# can download a pretrained extractor from:
+# http://kaldi-asr.org/models/12/0012_diarization_v1.tar.gz
+# and extract it.
+
+. ./cmd.sh
+. ./path.sh
+set -euo pipefail
+mfccdir=`pwd`/mfcc
+
+stage=0
+overlap_stage=0
+diarizer_stage=0
+nj=50
+decode_nj=15
+
+model_dir=exp/xvector_nnet_1a
+
+train_set=train
+test_sets="dev eval"
+
+diarizer_type=spectral  # must be one of (ahc, spectral, vbx)
+
+. utils/parse_options.sh
+
+# Path where AMI gets downloaded (or where locally available):
+AMI_DIR=$PWD/wav_db # Default,
+case $(hostname -d) in
+  fit.vutbr.cz) AMI_DIR=/mnt/matylda5/iveselyk/KALDI_AMI_WAV ;; # BUT,
+  clsp.jhu.edu) AMI_DIR=/export/corpora5/amicorpus ;; # JHU,
+  cstr.ed.ac.uk) AMI_DIR= ;; # Edinburgh,
+esac
+
+# Download AMI corpus, You need around 130GB of free space to get whole data ihm+mdm,
+if [ $stage -le 1 ]; then
+  if [ -d $AMI_DIR ] && ! touch $AMI_DIR/.foo 2>/dev/null; then
+    echo "$0: directory $AMI_DIR seems to exist and not be owned by you."
+    echo " ... Assuming the data does not need to be downloaded.  Please use --stage 1 or more."
+    exit 1
+  fi
+  if [ -e data/local/downloads/wget_$mic.sh ]; then
+    echo "data/local/downloads/wget_$mic.sh already exists, better quit than re-download... (use --stage N)"
+    exit 1
+  fi
+  local/ami_download.sh $mic $AMI_DIR
+fi
+
+# Prepare data directories.
+if [ $stage -le 2 ]; then
+  if ! [ -d data/local/annotations ]; then
+    local/ami_text_prep.sh data/local/downloads
+  fi
+
+  for dataset in train $test_sets; do
+    echo "$0: preparing $dataset set.."
+    mkdir -p data/$dataset
+    local/prepare_data.py data/local/annotations/${dataset}.txt \
+      $AMI_DIR data/$dataset
+    local/convert_rttm_to_utt2spk_and_segments.py --append-reco-id-to-spkr=true data/$dataset/rttm.annotation \
+      <(awk '{print $2" "$2" "$3}' data/$dataset/rttm.annotation |sort -u) \
+      data/$dataset/utt2spk data/$dataset/segments
+
+    # For the test sets we create dummy segments and utt2spk files using oracle speech marks
+    if ! [ $dataset == "train" ]; then
+      local/get_all_segments.py data/$dataset/rttm.annotation > data/$dataset/segments
+      awk '{print $1,$2}' data/$dataset/segments > data/$dataset/utt2spk
+    fi
+
+    utils/utt2spk_to_spk2utt.pl data/$dataset/utt2spk > data/$dataset/spk2utt
+    utils/fix_data_dir.sh data/$dataset
+  done
+fi
+
+# Feature extraction
+if [ $stage -le 3 ]; then
+  for dataset in train $test_sets; do
+    steps/make_mfcc.sh --mfcc-config conf/mfcc_hires.conf --nj $nj --cmd "$train_cmd" data/$dataset
+    steps/compute_cmvn_stats.sh data/$dataset
+    utils/fix_data_dir.sh data/$dataset
+  done
+fi
+
+if [ $stage -le 4 ]; then
+  echo "$0: preparing a AMI training data to train PLDA model"
```

Comment:
Perhaps here some more documentation is needed. I think you may add something about what this step is for. (feature norm)